src/Integration.UnitTests/LocalServices/ProjectSystemHelperTests.cs

```diff
@@ -618,6 +618,32 @@ public void ProjectSystemHelper_IsSolutionFullyLoaded_PropertyIsNotBoolean_Retur
             result.Should().BeFalse();
         }
 
+        [TestMethod]
+        public void ProjectSystemHelper_IsLegacyProject_NotLegacy_ReturnsFalse()
+        {
+            // Arrange
+            var mockProject = this.solutionMock.AddOrGetProject("dummy.proj", isLegacy: false);
+
+            // Act
+            var result = this.testSubject.IsLegacyProjectSystem(mockProject);
+
+            // Assert
+            result.Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void ProjectSystemHelper_IsLegacyProject_Legacy_ReturnsTrue()
+        {
+            // Arrange
+            var mockProject  = this.solutionMock.AddOrGetProject("dummy.proj", isLegacy: true);
+
+            // Act
+            var result = this.testSubject.IsLegacyProjectSystem(mockProject);
+
+            // Assert
+            result.Should().BeTrue();
```

Comment:
Unit tests for the new behaviour - don't add the ruleset file to the project for the new .NET-SDK project system.

Reply:
but this behaviour (not adding the file) is not checked in this file, only the legacy project detection... so why this comment here?

Reply:
My mistake - these tests are just for `IsLegacyProjectSystem`. The add/don't add tests are elsewhere.